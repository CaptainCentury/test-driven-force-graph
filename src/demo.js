import React from "react";
import styled from "styled-components";

import ForceGraph from "./components/force-graph";

const namesData = {
  nodes: [
    { name: "Adam" },
    { name: "Bob" },
    { name: "Carrie" },
    { name: "Donovan" },
    { name: "Edward" },
    { name: "Felicity" },
    { name: "George" },
    { name: "Hannah" },
    { name: "Iris" },
    { name: "Jerry" },
  ],
  edges: [
    { source: 0, target: 1 },
    { source: 0, target: 2 },
    { source: 0, target: 3 },
    { source: 0, target: 4 },
    { source: 1, target: 5 },
    { source: 2, target: 5 },
    { source: 2, target: 5 },
    { source: 3, target: 4 },
    { source: 5, target: 8 },
    { source: 5, target: 9 },
    { source: 6, target: 7 },
    { source: 7, target: 8 },
    { source: 8, target: 9 },
  ],
};

const AbcData = {
  nodes: [{ name: "A" }, { name: "B" }, { name: "C" }],
  edges: [
    { source: 0, target: 1 },
    { source: 0, target: 2 },
    { source: 1, target: 2 },
  ],
};

const BasicButton = styled.button`
  background: blue;
  border-radius: 3px;
  border: none;
  color: white;
`;

const DataButton = styled(BasicButton)`
  background: green;
`;

export const Demo = () => {
  const [data, setData] = React.useState(namesData);

  return (
    <>
      <ForceGraph style={{ color: "blue" }} dataset={data} labelMode="labels">
        <DataButton
          onClick={() => {
            setData(namesData);
          }}
        >
          Names
        </DataButton>
        <DataButton
          onClick={() => {
            setData(AbcData);
          }}
        >
          ABC
        </DataButton>
      </ForceGraph>
    </>
  );
};
