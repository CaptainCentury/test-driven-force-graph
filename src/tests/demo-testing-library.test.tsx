/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { DataButton, Demo } from "../demo";
import React from "react";

describe("Data button", () => {
  it("to render correctly", () => {
    const { getByText } = render(<DataButton>Test</DataButton>);
    expect(getByText("Test")).toMatchSnapshot();
  });
});

describe("Demo component", () => {
  it("renders Abc dataset button", () => {
    const { getByText } = render(<Demo />);
    const element = getByText("ABC");
    expect(element.tagName).toBe("BUTTON");
  });

  it("renders Names dataset button", () => {
    const { getByText } = render(<Demo />);
    const element = getByText(/names/i);
    expect(element.tagName).toBe("BUTTON");
  });

  it("renders correct number of buttons", () => {
    const { getAllByRole } = render(<Demo />);
    const elements = getAllByRole("button");
    expect(elements.length).toBe(2);
  });

  it("renders element Donovan in the graph", () => {
    const { getByText } = render(<Demo />);
    const element = getByText("Donovan");
    expect(element.parentElement.tagName).toBe("g");
  });

  it("switches data when data button is clicked", async () => {
    const user = userEvent.setup();
    const { getByText, queryByText } = render(<Demo />);
    const abcButton = getByText(/abc/i);
    const namesButton = getByText(/names/i);

    await user.click(abcButton);
    expect(queryByText("Donovan")).toBe(null);
    expect(queryByText("C").parentElement.tagName).toBe("g");

    await user.click(namesButton);
    expect(queryByText("C")).toBe(null);
    expect(queryByText("Donovan").parentElement.tagName).toBe("g");
  });

  it("styles ABC dataset button", () => {
    const { getByText } = render(<Demo />);
    const abcButton = getByText(/abc/i);
    expect(abcButton).toHaveStyleRule("background", "green");
    expect(abcButton).toHaveStyleRule("border-radius", "3px");
  });
});
