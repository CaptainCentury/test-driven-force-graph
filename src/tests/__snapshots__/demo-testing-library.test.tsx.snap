// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data button to render correctly 1`] = `
.c0 {
  background: blue;
  border-radius: 3px;
  border: none;
  color: white;
  margin: 2px;
}

.c1 {
  background: green;
}

<button
  class="c0 c1"
>
  Test
</button>
`;

exports[`Demo component to render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  background: blue;
  border-radius: 3px;
  border: none;
  color: white;
  margin: 2px;
}

.c1 {
  background: green;
}

<body>
    <div>
      <h1>
        ForceGraph example
      </h1>
      <figure
        style="color: blue;"
      >
        <svg
          height="360"
          style="border-style: solid; border-width: 1px;"
          width="360"
        >
          <g
            transform="translate(30,30)"
          >
            <line
              style="stroke: #ccc; stroke-width: 1;"
            />
            <line
              style="stroke: #ccc; stroke-width: 1;"
            />
            <line
              style="stroke: #ccc; stroke-width: 1;"
            />
            <line
              style="stroke: #ccc; stroke-width: 1;"
            />
            <line
              style="stroke: #ccc; stroke-width: 1;"
            />
            <line
              style="stroke: #ccc; stroke-width: 1;"
            />
            <line
              style="stroke: #ccc; stroke-width: 1;"
            />
            <line
              style="stroke: #ccc; stroke-width: 1;"
            />
            <line
              style="stroke: #ccc; stroke-width: 1;"
            />
            <line
              style="stroke: #ccc; stroke-width: 1;"
            />
            <line
              style="stroke: #ccc; stroke-width: 1;"
            />
            <line
              style="stroke: #ccc; stroke-width: 1;"
            />
            <line
              style="stroke: #ccc; stroke-width: 1;"
            />
            <g>
              <circle
                r="30"
                style="fill: #1f77b4;"
              />
              <text
                class="nodelabel"
                dy="0.3em"
                fill="#fff"
                pointer-events="none"
                stoke="none"
                style="font-size: 60px;"
                text-anchor="middle"
              >
                Adam
              </text>
            </g>
            <g>
              <circle
                r="30"
                style="fill: #ff7f0e;"
              />
              <text
                class="nodelabel"
                dy="0.3em"
                fill="#fff"
                pointer-events="none"
                stoke="none"
                style="font-size: 60px;"
                text-anchor="middle"
              >
                Bob
              </text>
            </g>
            <g>
              <circle
                r="30"
                style="fill: #2ca02c;"
              />
              <text
                class="nodelabel"
                dy="0.3em"
                fill="#fff"
                pointer-events="none"
                stoke="none"
                style="font-size: 60px;"
                text-anchor="middle"
              >
                Carrie
              </text>
            </g>
            <g>
              <circle
                r="30"
                style="fill: #d62728;"
              />
              <text
                class="nodelabel"
                dy="0.3em"
                fill="#fff"
                pointer-events="none"
                stoke="none"
                style="font-size: 60px;"
                text-anchor="middle"
              >
                Donovan
              </text>
            </g>
            <g>
              <circle
                r="30"
                style="fill: #9467bd;"
              />
              <text
                class="nodelabel"
                dy="0.3em"
                fill="#fff"
                pointer-events="none"
                stoke="none"
                style="font-size: 60px;"
                text-anchor="middle"
              >
                Edward
              </text>
            </g>
            <g>
              <circle
                r="30"
                style="fill: #8c564b;"
              />
              <text
                class="nodelabel"
                dy="0.3em"
                fill="#fff"
                pointer-events="none"
                stoke="none"
                style="font-size: 60px;"
                text-anchor="middle"
              >
                Felicity
              </text>
            </g>
            <g>
              <circle
                r="30"
                style="fill: #e377c2;"
              />
              <text
                class="nodelabel"
                dy="0.3em"
                fill="#fff"
                pointer-events="none"
                stoke="none"
                style="font-size: 60px;"
                text-anchor="middle"
              >
                George
              </text>
            </g>
            <g>
              <circle
                r="30"
                style="fill: #7f7f7f;"
              />
              <text
                class="nodelabel"
                dy="0.3em"
                fill="#fff"
                pointer-events="none"
                stoke="none"
                style="font-size: 60px;"
                text-anchor="middle"
              >
                Hannah
              </text>
            </g>
            <g>
              <circle
                r="30"
                style="fill: #bcbd22;"
              />
              <text
                class="nodelabel"
                dy="0.3em"
                fill="#fff"
                pointer-events="none"
                stoke="none"
                style="font-size: 60px;"
                text-anchor="middle"
              >
                Iris
              </text>
            </g>
            <g>
              <circle
                r="30"
                style="fill: #17becf;"
              />
              <text
                class="nodelabel"
                dy="0.3em"
                fill="#fff"
                pointer-events="none"
                stoke="none"
                style="font-size: 60px;"
                text-anchor="middle"
              >
                Jerry
              </text>
            </g>
          </g>
        </svg>
        <figcaption>
          <div>
            <button
              class="c0 c1"
            >
              Names
            </button>
            <button
              class="c0 c1"
            >
              ABC
            </button>
            <button
              class="c0 c1"
            >
              Yarnball
            </button>
          </div>
        </figcaption>
      </figure>
      <p>
        Node radius:
        <input
          id="radius"
          min="5"
          step="1"
          type="number"
          value="30"
        />
      </p>
      <p>
        Bond strength:
        <input
          id="link-strength"
          min="0.001"
          step="0.01"
          type="number"
          value="0.5"
        />
      </p>
    </div>
  </body>,
  "container": .c0 {
  background: blue;
  border-radius: 3px;
  border: none;
  color: white;
  margin: 2px;
}

.c1 {
  background: green;
}

<div>
    <h1>
      ForceGraph example
    </h1>
    <figure
      style="color: blue;"
    >
      <svg
        height="360"
        style="border-style: solid; border-width: 1px;"
        width="360"
      >
        <g
          transform="translate(30,30)"
        >
          <line
            style="stroke: #ccc; stroke-width: 1;"
          />
          <line
            style="stroke: #ccc; stroke-width: 1;"
          />
          <line
            style="stroke: #ccc; stroke-width: 1;"
          />
          <line
            style="stroke: #ccc; stroke-width: 1;"
          />
          <line
            style="stroke: #ccc; stroke-width: 1;"
          />
          <line
            style="stroke: #ccc; stroke-width: 1;"
          />
          <line
            style="stroke: #ccc; stroke-width: 1;"
          />
          <line
            style="stroke: #ccc; stroke-width: 1;"
          />
          <line
            style="stroke: #ccc; stroke-width: 1;"
          />
          <line
            style="stroke: #ccc; stroke-width: 1;"
          />
          <line
            style="stroke: #ccc; stroke-width: 1;"
          />
          <line
            style="stroke: #ccc; stroke-width: 1;"
          />
          <line
            style="stroke: #ccc; stroke-width: 1;"
          />
          <g>
            <circle
              r="30"
              style="fill: #1f77b4;"
            />
            <text
              class="nodelabel"
              dy="0.3em"
              fill="#fff"
              pointer-events="none"
              stoke="none"
              style="font-size: 60px;"
              text-anchor="middle"
            >
              Adam
            </text>
          </g>
          <g>
            <circle
              r="30"
              style="fill: #ff7f0e;"
            />
            <text
              class="nodelabel"
              dy="0.3em"
              fill="#fff"
              pointer-events="none"
              stoke="none"
              style="font-size: 60px;"
              text-anchor="middle"
            >
              Bob
            </text>
          </g>
          <g>
            <circle
              r="30"
              style="fill: #2ca02c;"
            />
            <text
              class="nodelabel"
              dy="0.3em"
              fill="#fff"
              pointer-events="none"
              stoke="none"
              style="font-size: 60px;"
              text-anchor="middle"
            >
              Carrie
            </text>
          </g>
          <g>
            <circle
              r="30"
              style="fill: #d62728;"
            />
            <text
              class="nodelabel"
              dy="0.3em"
              fill="#fff"
              pointer-events="none"
              stoke="none"
              style="font-size: 60px;"
              text-anchor="middle"
            >
              Donovan
            </text>
          </g>
          <g>
            <circle
              r="30"
              style="fill: #9467bd;"
            />
            <text
              class="nodelabel"
              dy="0.3em"
              fill="#fff"
              pointer-events="none"
              stoke="none"
              style="font-size: 60px;"
              text-anchor="middle"
            >
              Edward
            </text>
          </g>
          <g>
            <circle
              r="30"
              style="fill: #8c564b;"
            />
            <text
              class="nodelabel"
              dy="0.3em"
              fill="#fff"
              pointer-events="none"
              stoke="none"
              style="font-size: 60px;"
              text-anchor="middle"
            >
              Felicity
            </text>
          </g>
          <g>
            <circle
              r="30"
              style="fill: #e377c2;"
            />
            <text
              class="nodelabel"
              dy="0.3em"
              fill="#fff"
              pointer-events="none"
              stoke="none"
              style="font-size: 60px;"
              text-anchor="middle"
            >
              George
            </text>
          </g>
          <g>
            <circle
              r="30"
              style="fill: #7f7f7f;"
            />
            <text
              class="nodelabel"
              dy="0.3em"
              fill="#fff"
              pointer-events="none"
              stoke="none"
              style="font-size: 60px;"
              text-anchor="middle"
            >
              Hannah
            </text>
          </g>
          <g>
            <circle
              r="30"
              style="fill: #bcbd22;"
            />
            <text
              class="nodelabel"
              dy="0.3em"
              fill="#fff"
              pointer-events="none"
              stoke="none"
              style="font-size: 60px;"
              text-anchor="middle"
            >
              Iris
            </text>
          </g>
          <g>
            <circle
              r="30"
              style="fill: #17becf;"
            />
            <text
              class="nodelabel"
              dy="0.3em"
              fill="#fff"
              pointer-events="none"
              stoke="none"
              style="font-size: 60px;"
              text-anchor="middle"
            >
              Jerry
            </text>
          </g>
        </g>
      </svg>
      <figcaption>
        <div>
          <button
            class="c0 c1"
          >
            Names
          </button>
          <button
            class="c0 c1"
          >
            ABC
          </button>
          <button
            class="c0 c1"
          >
            Yarnball
          </button>
        </div>
      </figcaption>
    </figure>
    <p>
      Node radius:
      <input
        id="radius"
        min="5"
        step="1"
        type="number"
        value="30"
      />
    </p>
    <p>
      Bond strength:
      <input
        id="link-strength"
        min="0.001"
        step="0.01"
        type="number"
        value="0.5"
      />
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
